src/lib.rs

```diff
@@ -162,7 +162,11 @@ where
         while buf_left > 0 {
             let nb_bytes_readable = self.tokens_available().0.unwrap().min(buf_left);
             if nb_bytes_readable < readlimit.min(buf_left) {
-                std::thread::sleep(*window_time);
+                if let Some(lrc) = self.last_read_check {
+                    std::thread::sleep((*window_time - lrc.elapsed()).max(Duration::from_nanos(0)));
```

Comment:
It's not better to use a `saturating_sub` to block the overflow instead of a cmp ? Because is it overflow too much that it goes back to number upper than 0, it could be a problem ?
(Same for write)

Reply:
Actually no, it will only go to negative value. It will only overflow if we go under `i64::MIN`
(See [the code in std::time::Duration](https://docs.rs/time/latest/src/time/duration.rs.html#129))
But it's better to use `saturating_sub` than `max` anyway :+1:

Reply:
ahhh ok ye.